pkg/dao/snapshots.go

```diff
@@ -219,17 +207,36 @@ func (sDao *snapshotDaoImpl) FetchModel(ctx context.Context, uuid string, includ
 		result = sDao.db.WithContext(ctx).Where("uuid = ?", uuid).First(&snap)
 	}
 	if result.Error != nil {
-		if errors.Is(result.Error, gorm.ErrRecordNotFound) {
-			return models.Snapshot{}, &ce.DaoError{
-				Message:  "Could not find snapshot with UUID " + uuid,
-				NotFound: true,
-			}
-		}
-		return models.Snapshot{}, result.Error
+		return models.Snapshot{}, SnapshotsDBToApiError(result.Error, &uuid)
 	}
 	return snap, nil
 }
 
+func SnapshotsDBToApiError(e error, uuid *string) *ce.DaoError {
+	if e == nil {
+		return nil
+	}
+
+	daoError := ce.DaoError{}
+	if errors.Is(e, gorm.ErrRecordNotFound) {
+		msg := "Task not found"
+		if uuid != nil {
+			msg = fmt.Sprintf("Could not find snapshot with UUID %s", *uuid)
```

Comment:
nitpicky but i think just to be consistent with the others, the error message here could be `Snapshot with UUID %s not found`